packages/devextreme/js/ui/html_editor.d.ts

```diff
@@ -46,7 +46,109 @@ export type HtmlEditorImageUploadTab = 'url' | 'file';
 /** @public */
-export type HtmlEditorPredefinedToolbarItem = 'background' | 'bold' | 'color' | 'font' | 'italic' | 'link' | 'image' | 'size' | 'strike' | 'subscript' | 'superscript' | 'underline' | 'blockquote' | 'header' | 'increaseIndent' | 'decreaseIndent' | 'orderedList' | 'bulletList' | 'alignLeft' | 'alignCenter' | 'alignRight' | 'alignJustify' | 'codeBlock' | 'variable' | 'separator' | 'undo' | 'redo' | 'clear' | 'cellProperties' | 'tableProperties' | 'insertTable' | 'insertHeaderRow' | 'insertRowAbove' | 'insertRowBelow' | 'insertColumnLeft' | 'insertColumnRight' | 'deleteColumn' | 'deleteRow' | 'deleteTable';
+export type HtmlEditorPredefinedToolbarItem = 'background' | 'bold' | 'color' | 'font' | 'italic' | 'link' | 'image' | 'size' | 'strike' | 'subscript' | 'superscript' | 'underline' | 'blockquote' | 'header' | 'increaseIndent' | 'decreaseIndent' | 'orderedList' | 'bulletList' | 'alignLeft' | 'alignCenter' | 'alignRight' | 'alignJustify' | 'codeBlock' | 'variable' | 'separator' | 'undo' | 'redo' | 'clear' | 'cellProperties' | 'tableProperties' | 'insertTable' | 'insertHeaderRow' | 'insertRowAbove' | 'insertRowBelow' | 'insertColumnLeft' | 'insertColumnRight' | 'deleteColumn' | 'deleteRow' | 'deleteTable' | 'ai';
+/** @public */
+export type HtmlEditorAICommandName = 'summarize' | 'proofread' | 'expand' | 'shorten' | 'changeStyle' | 'changeTone' | 'translate' | 'askAI';
+/** @public */
+export type HtmlEditorAISimpleCommandName = Extract<HtmlEditorAICommandName, 'summarize' | 'proofread' | 'expand' | 'shorten' | 'askAI'>;
+/** @public */
+export type HtmlEditorAIChangeStyleOption =
+    | 'formal'
+    | 'informal'
+    | 'technical'
+    | 'business'
+    | 'creative'
+    | 'journalistic'
+    | 'academic'
+    | 'persuasive'
+    | 'narrative'
+    | 'expository'
+    | 'descriptive'
+    | 'conversational';
+/** @public */
+export type HtmlEditorAIChangeToneOption =
+    | 'professional'
+    | 'casual'
+    | 'straightforward'
+    | 'confident'
+    | 'friendly';
+/** @public */
+export type HtmlEditorAITranslateOption =
+    | 'arabic'
+    | 'chinese'
+    | 'english'
+    | 'french'
+    | 'german'
+    | 'japanese'
+    | 'spanish';
+
+/**
+ * @namespace DevExpress.ui.dxHtmlEditor
+ */
+export interface HtmlEditorAICommandBase {
+    name: HtmlEditorAICommandName | 'custom';
+    text?: string;
+}
+
+/**
+ * @namespace DevExpress.ui.dxHtmlEditor
+ */
+export interface HtmlEditorAISimpleCommand extends HtmlEditorAICommandBase {
+    name: HtmlEditorAISimpleCommandName;
+}
+
+/**
+ * @namespace DevExpress.ui.dxHtmlEditor
+ */
+export interface HtmlEditorAIChangeStyleCommand extends HtmlEditorAICommandBase {
+    name: 'changeStyle';
+    options?: HtmlEditorAIChangeStyleOption[];
```

Comment:
Could we remove code duplication, for example by using the Basic interface using generics? Something like this? 
```suggestion
export interface HtmlEditorAICommandBase<
    T extends HtmlEditorAICommandName,
    Options = undefined
> {
    name: T;
    text?: string;
    options?: Options;
}
```

Reply:
Updated in: https://github.com/DevExpress/DevExtreme/pull/29415/commits/054b856a8651904a8dce3fa6781c49cdb63625cd